### DISCUSSION TITLE
Store diff results to server

### DISCUSSION THREAD

**Discussion by geeksmith** [Ericsson/codechecker#4378] [Q&A]
I have a CodeChecker server with baseline runs. If I run analysis locally, then diff my results to the server I can generate nice reports in different formats of all the defects I've introduced.

The server's interface is much better than the local report interface, so I'd like to view my new defects on the server. However, I don't want to upload all the defects found, just the new ones. Is there a way to store only the new defects on the server in a separate run from the baseline?

**Comment by bruntib** [comment]
Hi,

No, it is not possible to upload only the new reports, and it wouldn't give proper results. If run "A" contains a baseline analysis and run "B" contains only the new reports, then a diff between these two would show all other reports as resolved, since they are stored in "A" but not in "B".

Consider storing the new analysis to the same run. This will present the new results natively. It is a good workflow to store the analysis reports of the same branch to the same run and that way you can monitor the lifecycle of reports easily.

If you plan to store only the diff in another run because of efficiency reasons, then I can tell that we're working on the optimization of the storage which can be a slow operation in case of huge report directories.

**Comment by geeksmith** [comment]
My use case is for continuous integration test results. I have baseline runs that I update regularly. All unmerged changes go through a verification pipeline that includes running CodeChecker against the changed files. I intend to upload the CI analysis results as individual runs, highlighting any new defects. However, I can't upload only the new defects, I can only upload the entire results set.

I realized (after submitting this question) that as long as the runs are within the same product, the known issues should be easy to filter out in the web interface, especially if all existing issues have all been reviewed. In that case, CI runs would only show new defects. Correct?

**Comment by bruntib** [comment]
The intended workflow in this case is to use `CodeChecker cmd diff --new -b <baseline> -n <changes>`. This command prints only the new reports compared to the baseline (there are also variants for printing `--resolved` and `--unresolved` reports instead of `--new`). The `<baseline>` and `<changes>` can be either a URL of the run stored to the web server: `localhost:8001/Default/<run_name>` or a report directory. When CodeChecker is integrated to a CI system, then the reports of the new, changed code are usually not stored to the server, but used only for comparison to the baseline.

**Comment by geeksmith** [comment]
That makes sense. The webserver interface is so much better than the other means of viewing defects, so we'd like to leverage it for viewing results from unmerged code outside the baseline. I think I've got something that will work for this, thank you for the feedback.
